server/modules/surveyRdb/schemaRdb/dataView.js

```diff
@@ -23,91 +23,145 @@ export const columns = {
+ * (columns relative to its children and to its ancestors' children).
+ *
+ * @param {!object} survey - The survey.
+ * @param {!object} nodeDef - The context node definition.
+ * @returns {Array} List of column names.
+ */
+export const getNodeDefColumnNames = (survey, nodeDef) => {
+  const getNodeDefCols = (nodeDefCurrent, isSelf) => {
+    const nodeDefColumns = R.pipe(
+      R.map(DataCol.getNames),
+      R.flatten
+    )(DataTable.getNodeDefColumns(survey, nodeDefCurrent))
+
+    // If is not root, prepend parent uuid
+    if (!NodeDef.isRoot(nodeDefCurrent)) {
+      nodeDefColumns.unshift(getColUuid(Survey.getNodeDefParent(nodeDefCurrent)(survey)))
+    }
+
+    // If nodeDef isSelf (starting nodeDef) prepend col uuid
+    if (isSelf) {
+      nodeDefColumns.unshift(getColUuid(nodeDefCurrent))
+    }
+    return nodeDefColumns
+  }
 
-export const getSelectFields = (survey, nodeDef, resultStepViews) => {
-  if (NodeDef.isVirtual(nodeDef)) {
-    return ['*']
+  const colNames = []
+
+  Survey.visitAncestorsAndSelf(nodeDef, (nodeDefCurrent) => {
+    // Do not include node defs of calculation steps
+    const cols = getNodeDefCols(nodeDefCurrent, NodeDef.isEqual(nodeDefCurrent)(nodeDef))
+    colNames.unshift(...cols)
+  })(survey)
+
+  return colNames
+}
+
+/**
+ * Generates a list of fields to be used in the projection of the data view.
+ * The fields will be related to the node definition specified
+ * (columns related to its children definitions or to its ancestors children definitions).
+ *
+ * @param {!object} survey - The context survey.
+ * @param {!object} nodeDef - The node definition.
+ * @returns {Array} List of fields.
+ */
+const _getSelectFieldsNodeDefs = (survey, nodeDef) => {
+  const getSelectFieldNodeDef = (nodeDefCurrent, isSelf) => {
+    const fieldsNodeDef = R.pipe(
+      R.map(DataCol.getNames),
+      R.flatten,
+      R.map((name) => `${isSelf ? alias : aliasParent}.${name}`)
+    )(DataTable.getNodeDefColumns(survey, nodeDefCurrent))
+
+    // If is not root, prepend parent uuid
+    if (!NodeDef.isRoot(nodeDefCurrent)) {
+      fieldsNodeDef.unshift(`${aliasParent}.${getColUuid(Survey.getNodeDefParent(nodeDefCurrent)(survey))}`)
+    }
+
+    // If nodeDef isSelf (starting nodeDef) prepend col uuid
+    if (isSelf) {
+      fieldsNodeDef.unshift(`${alias}.${DataTable.colNameUuuid} as ${getColUuid(nodeDefCurrent)}`)
+    }
+
+    return fieldsNodeDef
   }
 
   const fields = []
 
-  Survey.visitAncestorsAndSelf(nodeDef, nodeDefCurrent => {
+  Survey.visitAncestorsAndSelf(nodeDef, (nodeDefCurrent) => {
     // Do not include node defs of calculation steps
-    const cols = getCols(survey, nodeDefCurrent, NodeDef.isEqual(nodeDefCurrent)(nodeDef))
-    fields.unshift(...cols)
+    const fieldsNodeDef = getSelectFieldNodeDef(nodeDefCurrent, NodeDef.isEqual(nodeDefCurrent)(nodeDef))
+    fields.unshift(...fieldsNodeDef)
   })(survey)
 
+  return fields
+}
```

Comment:
this is very similar to getNodeDefColumnNames. 
can we just use getNodeDefColumnNames and remove _getSelectFieldsNodeDefs?
is the only difference the alias? if that's the case, we can create an inner function 
_getNodeDefColumnNames = (survey, nodeDef, alias=null, aliasParent = null) ?

Reply:
ok, good